hoomd/md/improper.py

```diff
@@ -110,3 +110,46 @@ def __init__(self):
                 chi0=hoomd.data.typeconverter.nonnegative_real,
                 len_keys=1))
         self._add_typeparam(params)
+
+class Periodic(Improper):
+    """Periodic improper force.
+
+    `Periodic` computes forces, virials, and energies on all impropers in the
+    simulation state with:
+
+    .. math::
+
+        U(phi) = \\k / 2 \\left( 1 + d * cos(n * \\phi - \\phi_{0})  \\right )
+
+    Attributes:
+        params(`TypeParameter` [``improper type``, `dict`]):
+            The parameter of the harmonic impropers for each improper type. The
+            dictionary has the following keys:
+
+            * ``k`` (`float`, **required**), potential constant :math:`k`
+              :math:`[\\mathrm{energy}]`.
+            * ``chi{0}`` (`float`, **required**), equilibrium angle
+              :math:`\\phi_0` :math:`[\\mathrm{radian}]`.
+            * ``n`` (`float`, **required**), periodic number
+              :math:`\\n` :math:`[\\mathrm{dimensionless}]`.
+            * ``d`` (`float`, **required**), sign factor
+              :math:`\\n` :math:`[\\mathrm{dimensionless}]`.
+
+    Example::
+
+        harmonic = hoomd.md.improper.Periodic()
+        harmonic.params['A-B-C-D'] = dict(k=1.0, n = 1, phi0=0)
```

Comment:
Make these changes to test the example code snippet using [sybil](https://sybil.readthedocs.io).

```suggestion
    .. rubric:: Example:

    .. code-block:: python

        periodic = hoomd.md.improper.Periodic()
        periodic.params['A-B-C-D'] = dict(k=1.0, n = 1, phi0=0)
        
        simulation.integrator.forces = [periodic]
```
and set the parameters appropriately.

Also, add the following to the top of the file in the module level docstring:
```
.. skip: next if(not hoomd.version.md_built)

.. invisible-code-block: python

    if hoomd.version.md_built:
        simulation = hoomd.util.make_example_simulation()
        simulation.operations.integrator = hoomd.md.Integrator(dt=0.001)
```

Reply:
> Make these changes to test the example code snippet using [sybil](https://sybil.readthedocs.io).
> 
> and set the parameters appropriately.
> 
> Also, add the following to the top of the file in the module level docstring:
> 
> ```
> .. skip: next if(not hoomd.version.md_built)
> 
> .. invisible-code-block: python
> 
>     if hoomd.version.md_built:
>         simulation = hoomd.util.make_example_simulation()
>         simulation.operations.integrator = hoomd.md.Integrator(dt=0.001)
> ```

This example code produces the error:

```AttributeError: 'Simulation' object has no attribute 'integrator' ```

Do you have a suggestion for a different example case?

Reply:
You can find many examples scattered throughout the HOOMD documentation, search for `.. invisible-code-block: python`. I am certain this code works. Here is a test:
```
>>> simulation = hoomd.util.make_example_simulation()
>>> simulation.operations.integrator = hoomd.md.Integrator(dt=0.001)
>>>
```